crates/uv/src/commands/project/remove.rs

```diff
@@ -0,0 +1,92 @@
+use anyhow::Result;
+use std::str::FromStr;
+use uv_distribution::pyproject_mut::PyProjectTomlMut;
+
+use distribution_types::IndexLocations;
+use pep508_rs::Requirement;
+use pypi_types::LenientRequirement;
+use uv_cache::Cache;
+use uv_configuration::{ExtrasSpecification, PreviewMode, Upgrade};
+use uv_distribution::ProjectWorkspace;
+use uv_warnings::warn_user;
+
+use crate::commands::{project, ExitStatus};
+use crate::printer::Printer;
+
+/// Remove one or more packages from the project requirements.
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn remove(
+    requirements: Vec<String>,
+    preview: PreviewMode,
+    cache: &Cache,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if preview.is_disabled() {
+        warn_user!("`uv remove` is experimental and may change without warning.");
+    }
+
+    // Find the project requirements.
+    let project = ProjectWorkspace::discover(&std::env::current_dir()?, None).await?;
+
+    let mut pyproject = PyProjectTomlMut::from_toml(project.current_project().pyproject_toml())?;
+    for req in requirements {
+        let req = Requirement::from(LenientRequirement::from_str(&req)?);
+        if pyproject.remove_dependency(&req)?.is_none() {
```

Comment:
What happens if you `uv remove anyio`, without including the specifier?

Reply:
`remove_dependency` removes by package name, so that works.